rust/src/io_utils.rs

```diff
@@ -4,20 +4,24 @@ use crc32fast::Hasher;
 
 pub struct CountingCrcWriter<W> {
```

Comment:
rather than making this optional, would it make sense to dynamically use or not use this CountingCrcWriter wrapper?